components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/main/java/org/wso2/carbon/identity/api/server/action/management/v1/core/ServerActionManagementService.java

```diff
@@ -0,0 +1,280 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.action.management.v1.core;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.action.mgt.exception.ActionMgtException;
+import org.wso2.carbon.identity.action.mgt.model.Action;
+import org.wso2.carbon.identity.action.mgt.model.AuthType;
+import org.wso2.carbon.identity.action.mgt.model.EndpointConfig;
+import org.wso2.carbon.identity.action.mgt.model.TypeEnums;
+import org.wso2.carbon.identity.api.server.action.management.common.ActionManagementServiceHolder;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionBasicResponse;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionModel;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionResponse;
+import org.wso2.carbon.identity.api.server.action.management.v1.ActionTypesResponseItem;
+import org.wso2.carbon.identity.api.server.action.management.v1.AuthenticationType;
+import org.wso2.carbon.identity.api.server.action.management.v1.Endpoint;
+import org.wso2.carbon.identity.api.server.action.management.v1.util.ActionMgtEndpointUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_CODE_ADD_ACTION;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_CODE_UPDATE_ACTION;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_NO_ACTION_CONFIGURED_ON_GIVEN_ID;
+
+/**
+ * Server Action Management Service.
+ */
+public class ServerActionManagementService {
+
+    private static final Log LOG = LogFactory.getLog(ServerActionManagementService.class);
+
+    /**
+     * Create a new action.
+     *
+     * @param actionType   Action type.
+     * @param actionModel  Action model.
+     * @return Action response.
+     */
+    public ActionResponse createAction(String actionType, ActionModel actionModel) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Adding Action for Action Type: " + actionType);
+        }
```

Comment:
Can we move this debug log to the java service layer.
The underlying service maybe invoked from different paths

Reply:
Fixed with https://github.com/wso2/identity-api-server/pull/626/commits/d3cd7cf52a0c9022aafe2dd2fdf10a0079124a59